dev/run-pip-tests

```diff
@@ -120,6 +122,10 @@ for python in "${PYTHON_EXECS[@]}"; do
     python "$FWDIR"/dev/pip-sanity-check.py
     echo "Run the tests for context.py"
     python "$FWDIR"/python/pyspark/context.py
+    if [ -n "$TEST_PYARROW" ]; then
+      echo "Run tests for pyarrow"
+      SPARK_TESTING=1 "$FWDIR"/bin/pyspark pyspark.sql.tests ArrowTests
```

Comment:
pyarrow has pip packages now, does that help? @BryanCutler I can't read your comment and understand why they required a conda env but maybe that's it?